src/components/user/login/index.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import {
+  Grid, Row, Col, Alert,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { loginAPI } from '../../../store/actions/login';
+import { ROUTES } from '../../../constants';
+import LoginForm from './login-form';
+
+class Login extends Component {
+  constructor() {
+    super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(fields) {
+    const { login } = this.props;
+    login(fields);
+  }
+
+  renderFormError() {
+    const { loginState } = this.props;
+    const { message } = loginState;
```

Comment:
What is this message? And, where does it come from?

Messages in your application should be defined in constants in your frontend and not be delivered from your backend. Having them in your frontend will allow you to:
1. interpolate variable data in the messages
2. handle accessibility concerns
3. handle i18n translations
4. optimize the payload from the backend
5. ensure the UI text is independent from the backend

Refactor this to have your error and success messages mapped to status codes from the API call payload.

Reply:
The message is coming from the backend. I never knew this... I always thought the point of including the message in the payload of the request sent from the back-end to the front-end was so we could re-use that same message and display it on the front-end component. Good to know~